### PR TITLE
feat(ecommerce-for-io): set x-user-id header

### PR DESCRIPTION
**Pull request by pietro-tota**
### List of changes
Set `x-client-id` header for eCommerce `POST transactions` v2 api valued with PDV userId token

### Motivation and context
Set `x-client-id` header for eCommerce `POST transactions` v2 api valued with PDV userId token

### Type of changes

- [ ] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### CODE REVIEW

File: src/domains/ecommerce-app/api/ecommerce-transactions-service/v2/_openapi.json.tpl [hunk 1]
```diff
@@ -41,6 +41,15 @@
         ],
         "description": "Create a new transaction activating the payments notice by meaning of 'Nodo' ActivatePaymentNotice primitive",
         "parameters": [
+          {
+            "in": "header",
+            "name": "X-Client-Id",
+            "required": true,
```
Comment (infantesimone) [inline src/domains/ecommerce-app/api/ecommerce-transactions-service/v2/_openapi.json.tpl:47]: required?
it is not required in https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/470

File: src/domains/ecommerce-app/api/ecommerce-transactions-service/v2/_openapi.json.tpl [hunk 2]
```diff
@@ -41,6 +41,15 @@
         ],
         "description": "Create a new transaction activating the payments notice by meaning of 'Nodo' ActivatePaymentNotice primitive",
         "parameters": [
+          {
```
Comment (infantesimone) [inline src/domains/ecommerce-app/api/ecommerce-transactions-service/v2/_openapi.json.tpl:44]: Why update this file?
These parameters are entered at the APIM level.

Comment (infantesimone) [reply src/domains/ecommerce-app/api/ecommerce-transactions-service/v2/_openapi.json.tpl:44]: this API are not currently used, but they will potentially be fields populated from an API key (user group or id) in the future

Comment (pietro-tota) [reply src/domains/ecommerce-app/api/ecommerce-transactions-service/v2/_openapi.json.tpl]: You're right :D rollbacked witth https://github.com/pagopa/pagopa-infra/pull/1908/commits/f5b8e38d58b2ac796c604263dfc5e637460af332

**Review by infantesimone — APPROVED** [review]

**Pull request merged**